cmd/zfs/zfs_iter.c

```diff
@@ -298,10 +305,15 @@ zfs_sort(const void *larg, const void *rarg, void *data)
 	zfs_sort_column_t *sc = (zfs_sort_column_t *)data;
 	zfs_sort_column_t *psc;
 
+	if (sc && sc->sc_fast_txg && l && r) {
```

Comment:
Can l or r really be NULL here?  I don't see zfs_get_user_props(), zfs_prop_get_int() or other places below checking them for NULL.  Same question is about sc, but I haven't traced it deep.

Reply:
You are right, 'l' and 'r' should not be NULL and AVL should be able to return with us with valid handles. However, 'sc' can be NULL if no sort column is specified, e.g. "zfs list -t snapshot -o name".